### PR TITLE
Add Dart Code Metrics

### PR DESCRIPTION
**Pull request by elliette**
Work towards https://github.com/dart-lang/webdev/issues/2053

Applies fixes for rules that are trivial to fix. Rules with non-trivial fixes are commented out, with TODO.

### CODE REVIEW

**Review by annagrin — APPROVED** [review]
LGTM!

File: dwds/analysis_options.yaml [hunk 1]
```diff
@@ -22,3 +22,45 @@ linter:
...
+    # - avoid-non-null-assertion
+    # - avoid-passing-async-when-sync-expected
+    - avoid-redundant-async
+    # - avoid-throw-in-catch-block
+    - avoid-unnecessary-type-assertions
+    - avoid-unnecessary-type-casts
+    - avoid-unrelated-type-assertions
+    - avoid-unused-parameters
+    - binary-expression-operand-order
+    - double-literal-format
```
Comment (jacob314) [inline dwds/analysis_options.yaml:51]: can you include all of the lint rules for DCM here with the ones you have not enabled commented out?

File: dwds/debug_extension_mv3/web/background.dart [hunk 2]
```diff
@@ -62,7 +62,7 @@ void _registerListeners() {
 Future<void> _handleRuntimeMessages(
   dynamic jsRequest,
   MessageSender sender,
-  Function sendResponse,
+  Function _,
```
Comment (jacob314) [inline dwds/debug_extension_mv3/web/background.dart:65]: not sure this change is an improvement. consider using an ignore comment instead. DCM should be able to figure out that the arg is needed based on where this is used as a listener.

File: dwds/debug_extension_mv3/web/cross_extension_communication.dart [hunk 3]
```diff
@@ -29,7 +29,7 @@ final _eventsForAngularDartDevTools = {
 
 Future<void> handleMessagesFromAngularDartDevTools(
   dynamic jsRequest,
-  MessageSender sender,
+  MessageSender _,
```
Comment (jacob314) [inline dwds/debug_extension_mv3/web/cross_extension_communication.dart:32]: consider an ignore comment here as well.

File: dwds/debug_extension_mv3/web/debug_session.dart [hunk 4]
```diff
@@ -234,9 +234,11 @@ _enableExecutionContextReporting(int tabId) {
       if (chromeError != null) {
         final errorMessage = _translateChromeError(chromeError.message);
         chrome.notifications.create(
-          /*notificationId*/ null,
+          /*notificationId*/
```
Comment (jacob314) [inline dwds/debug_extension_mv3/web/debug_session.dart:237]: same as before. Switch these to `// comment`

File: dwds/debug_extension_mv3/web/utils.dart [hunk 5]
```diff
@@ -42,7 +42,7 @@ Future<Tab?> getTab(int tabId) {
   return completer.future;
 }
 
-Future<Tab?> get activeTab async {
+Future<Tab?> get activeTab {
```
Comment (jacob314) [inline dwds/debug_extension_mv3/web/utils.dart:45]: i keep finding it interesting how many cases this lint catches.

File: dwds/lib/src/debugging/inspector.dart [hunk 6]
```diff
@@ -619,6 +616,12 @@ class AppInspector implements AppInspectorInterface {
     });
   }
 
+  Iterable<String> _userLibraryUris(Iterable<LibraryRef> libraries) {
```
Comment (jacob314) [inline dwds/lib/src/debugging/inspector.dart:619]: what lint caused this refactor?

**Review by jacob314 — COMMENTED** [review]
lgtm

File: dwds/analysis_options.yaml (hunk 1 above)
Comment (elliette) [reply dwds/analysis_options.yaml:51]: Added changes for this (and other PR comments) at https://github.com/dart-lang/webdev/pull/2072

File: dwds/lib/src/debugging/inspector.dart (hunk 6 above)
Comment (elliette) [reply dwds/lib/src/debugging/inspector.dart:619]: Ah I think this snuck in when I was trying to address:

```
metrics:
   cyclomatic-complexity: 20
   number-of-parameters: 5
   maximum-nesting-level: 5
```

before disabling those checks  as a follow-up. Forget to revert the changes here.

**Pull request merged**